Makefile

```diff
@@ -1,73 +1,89 @@
-include .env.dev
-
-ENV_DIR ?= venv
-ENV = $(CURDIR)/$(ENV_DIR)
-CELERY = $(ENV)/bin/celery
-PRECOMMIT = $(ENV)/bin/pre-commit
-PIP = $(ENV)/bin/pip
-PYTHON3 = $(ENV)/bin/python3
-PYTEST = $(ENV)/bin/pytest
-
-DOCKER_FILE ?= docker-compose-developer.yml
-
-define setup_engine_env
-	export `grep -v '^#' .env | xargs -0` && cd engine
+DOCKER_COMPOSE_FILE = docker-compose-developer.yml
+
+# if .env is not present simply touch it
+$(shell touch .env)
+include .env
+
+# if COMPOSE_PROFILES is set in .env use it, otherwise use a default (or what is passed in as an arg)
+ifeq ($(COMPOSE_PROFILES),)
+	COMPOSE_PROFILES=engine,oncall_ui,redis,grafana
+endif
+
+# conditionally assign DB based on what is present in COMPOSE_PROFILES
+ifeq ($(findstring mysql,$(COMPOSE_PROFILES)),mysql)
+	DB=mysql
+else ifeq ($(findstring mysql,$(COMPOSE_PROFILES)),mysql)
+	DB=postgres
+else
+	DB=sqlite
+endif
+
+# conditionally assign BROKER_TYPE based on what is present in COMPOSE_PROFILES
+ifeq ($(findstring redis,$(COMPOSE_PROFILES)),redis)
+	BROKER_TYPE=redis
```

Comment:
`redis` is always needed (at least for caching), so seems like it would be a good idea to allow overriding `BROKER_TYPE` to `rabbitmq` even if `redis` is in `COMPOSE_PROFILES`

Reply:
I think by inversing the logic to this, it _should_ work:
```Makefile
# conditionally assign BROKER_TYPE based on what is present in COMPOSE_PROFILES
# if the user specifies both rabbitmq and redis, we'll make the assumption that rabbitmq is the broker
ifeq ($(findstring rabbitmq,$(COMPOSE_PROFILES)),rabbitmq)
	BROKER_TYPE=rabbitmq
else
	BROKER_TYPE=redis
endif
```

Test cases:
```shell
❯ COMPOSE_PROFILES=redis,rabbitmq make stop
COMPOSE_PROFILES=redis,rabbitmq DB=sqlite BROKER_TYPE=rabbitmq docker-compose -f docker-compose-developer.yml down

❯ COMPOSE_PROFILES=rabbitmq make stop
COMPOSE_PROFILES=rabbitmq DB=sqlite BROKER_TYPE=rabbitmq docker-compose -f docker-compose-developer.yml down

❯ COMPOSE_PROFILES=redis make stop
COMPOSE_PROFILES=redis DB=sqlite BROKER_TYPE=redis docker-compose -f docker-compose-developer.yml down
```

I would argue that it's safe to assume that if the user specifies both `redis` and `rabbitmq` in `COMPOSE_PROFILES` that `rabbitmq` should be the broker

Reply:
Yes, this looks nice